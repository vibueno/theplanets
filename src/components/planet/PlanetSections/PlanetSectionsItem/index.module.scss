@use 'SRC/variables' as *;
@use 'SRC/mixins.scss' as mixins;

.planetSectionsBtn
{
  font: 700 9px/22px Spartan, Arial, Helvetica;

  display: flex;
  align-items: center;

  width: 100%;
  height: 48px;

  cursor: pointer;
  letter-spacing: 1.93px;
  text-transform: uppercase;

  border: none;
  background-color: $color-blue-dark;

  .planetSectionsBtnCaption
  {
    display: inline-block;
  }

  .planetSectionsBtnCaption,
  .planetSectionsBtnCaptionLong
  {
    color: $color-white;
  }


  .planetSectionsBtnCaptionLong
  {
    display: none;
  }
}


@media only screen and (max-width: 767px)
{
  .planetSectionsBtn
  {
    position: relative;

    justify-content: center;

    @include mixins.planet-sections-menu-selected;

    #overview
    {
      grid-column: 1 / 2;
    }

    #structure
    {
      grid-column: 2 / 3;
    }

    #geology
    {
      grid-column: 3 / 4;
    }

    &::after
    {
      position: absolute;
      bottom: 0;
      left: 0;

      display: block;

      width: 100%;
      height: 5px;

      content: '';
      transition: transform .3s ease;
      transform: scaleX(0);
    }
    &[data-selected='true']
    {
      .planetSectionsBtnCaption
      {
        color: $color-grey-light;
      }
    }
  }

  .planetSectionsBtnNumbering
  {
    display: none;
  }
}

@media only screen and (min-width: 768px)
{
  .planetSectionsBtn
  {
    font: 700 12px/22px Spartan, Arial, Helvetica;

    position: relative;

    justify-content: left;

    margin: 0 0 1em 0;
    padding: 5px 25px;

    border: none;

    @include mixins.planet-sections-btn-panel-selected();

    .planetSectionsBtnNumbering
    {
      display: inline-block;

      padding: 0 25px 0 0;
    }

    .planetSectionsBtnCaption
    {
      display: none;
    }

    .planetSectionsBtnCaptionLong
    {
      display: inline-block;
    }

    &[data-selected='false']
    {
      transition: background-color .3s;

      border: $border;

      &:hover
      {
        background-color: $color-grey-dark;
      }

      .planetSectionsBtnNumbering
      {
        color: $color-grey-light;
      }
    }
  }
}

@media only screen and (min-width: 1200px) and (max-height:767px)
{
  .planetSectionsBtn
  {
    height: 40px;

    &:last-child
    {
      margin: 0 0 .5em 0;
    }
  }
}
