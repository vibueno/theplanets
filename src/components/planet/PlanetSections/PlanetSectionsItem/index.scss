@use 'SRC/variables.scss' as *;
@use 'SRC/mixins.scss' as mixins;

.btn
{
  font: 700 9px/22px Spartan, Arial, Helvetica;

  display: flex;

  margin: 0;

  cursor: pointer;
  text-align: left;
  letter-spacing: 1.93px;
  text-transform: uppercase;

  border: none;
  background-color: $color-blue-dark;

  &[data-selected='false']
  {
    .btn-numbering
    {
      color: $color-grey-light;
    }
  }

  .btn-numbering
  {
    display: none;

    padding: 0 25px 0 0;
  }

  &[data-selected='false']
  {
    .btn-caption
    {
      color: $color-white;
    }
  }

  .btn-caption
  {
    display: inline-block;
  }

  .btn-caption-long
  {
    display: none;

    color: $color-white;
  }
}

.planet-sections-menu
{
  @include mixins.planet-sections-menu-selected;
}

@media only screen and (min-width: 768px)
{
  .planet-sections-btn-panel
  {
    .btn
    {
      .btn-caption
      {
        display: none;
      }

      .btn-caption-long
      {
        display: inline-block;
      }
    }
  }
}
