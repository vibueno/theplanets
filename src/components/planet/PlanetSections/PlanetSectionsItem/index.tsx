import React, { MouseEventHandler } from 'react';

type PlanetSectionsMenuItemProps = {
  sectionKey: string;
  currentSectionKey: string;
  className: string;
  caption: string;
  captionLong: string;
  classNameNumbering: string;
  numbering: string;
  clickHandler: (
    e: React.MouseEvent<HTMLButtonElement>,
    ref: React.RefObject<HTMLButtonElement>
  ) => void;
};

const PlanetSectionsMenuItem = ({
  sectionKey,
  currentSectionKey,
  className,
  caption,
  captionLong,
  classNameNumbering,
  numbering,
  clickHandler
}: PlanetSectionsMenuItemProps) => {
  const ref = React.useRef<HTMLButtonElement>(null);

  return (
    <button
      id={sectionKey}
      className={`btn-section ${className}`}
      onClick={e => clickHandler(e, ref)}
      data-selected={sectionKey === currentSectionKey}
      ref={ref}
    >
      <span className={classNameNumbering}>{numbering}</span>
      <span className="btn-caption">{caption}</span>
      <span className="btn-caption-long">{captionLong}</span>
    </button>
  );
};
export default PlanetSectionsMenuItem;
