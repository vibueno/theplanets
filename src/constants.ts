const sections = {
  overview: {
    id: 'overview',
    number: '01',
    btnCaption: 'overview',
    btnCaptionExtended: 'overview'
  },
  structure: {
    id: 'structure',
    number: '02',
    btnCaption: 'structure',
    btnCaptionExtended: 'internal structure'
  },
  geology: {
    id: 'geology',
    number: '03',
    btnCaption: 'geology',
    btnCaptionExtended: 'surface geology'
  }
};

const planetNames = {
  mercury: 'mercury',
  venus: 'venus',
  earth: 'earth',
  mars: 'mars',
  jupiter: 'jupiter',
  saturn: 'saturn',
  uranus: 'uranus',
  neptune: 'neptune'
};

const planetKeyData = {
  rotation: {
    id: 'rotation',
    title: 'rotation time'
  },
  revolution: {
    id: 'revolution',
    title: 'rotation time'
  },
  radius: {
    id: 'radius',
    title: 'radius'
  },
  temperature: {
    id: 'temperature',
    title: 'average temp.'
  }
};

export { sections, planetNames, planetKeyData };
