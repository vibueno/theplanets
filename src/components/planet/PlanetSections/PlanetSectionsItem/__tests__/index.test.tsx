import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import PlanetSectionsItem from '../';
import { PLANETS, SECTIONS } from 'SRC/constants';

describe('PlanetSectionsItem', () => {
  const initialState = {
    appState: {
      isMenuOpen: false,
      isTransitionDone: true,
      currentPlanetKey: PLANETS.MERCURY.KEY,
      currentSectionKey: SECTIONS.OVERVIEW.KEY
    }
  };

  const mockStore = configureStore();

  let store = mockStore(initialState);
  const props = {
    sectionKey: SECTIONS.OVERVIEW.KEY,
    currentSectionKey: SECTIONS.OVERVIEW.KEY,
    className: `planetSectionsBtn-${PLANETS[PLANETS.MERCURY.KEY].NAME}`,
    caption: 'caption',
    captionLong: 'captionLong',
    numbering: SECTIONS[SECTIONS.OVERVIEW.KEY].NUMBER,
    clickHandler: () => {}
  };

  it('PlanetSectionsItem renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PlanetSectionsItem {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('PlanetSectionsItem selected if currentPlanetKey === planetKey', () => {
    render(
      <Provider store={store}>
        <PlanetSectionsItem {...props} />
      </Provider>
    );
    expect(screen.queryByRole('button')).toHaveAttribute(
      'data-selected',
      'true'
    );
  });
});
