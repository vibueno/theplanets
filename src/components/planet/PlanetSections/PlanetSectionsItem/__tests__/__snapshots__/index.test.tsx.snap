// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetSectionsItem PlanetSectionsItem renders correctly 1`] = `
<button
  className="undefined undefined"
  data-selected={true}
  id="OVERVIEW"
  onClick={[Function]}
>
  <span>
    01
  </span>
  <span>
    caption
  </span>
  <span>
    captionLong
  </span>
</button>
`;
